src/Emulator/Peripherals/Peripherals/Timers/OpenTitan_Timer.cs

```diff
@@ -0,0 +1,344 @@
+                            Timers[timerEnablesIdx, i].Enabled = val;
+                        }
+                    });
+                }
+                this.Log(LogLevel.Noisy, "Add REG: CTRL[0x{0:X}]", address + offset);
+                registersMap.Add(address + offset, controlRegister);
+            }
+        }
+
+        private void AddTimerRegisters(Dictionary<long, DoubleWordRegister> registersMap, long address, int hartId)
+        {
+            var hartxOffset = address + MaxHartRegMapSize * (hartId + 1);
+            var compareOffset = AddTimerConfigAndValueRegisters(registersMap, hartxOffset, hartId);
+            var intrEnableOffset = AddTimerCompareRegisters(registersMap, compareOffset, hartId);
+            var intrStateOffset = AddMultiRegisters(registersMap, intrEnableOffset, hartId, numberOfTimers, 
+                flagPrefix: "IE", regName:"INTR_ENABLE", multiRegWriteCallback: (hartIdx, timerIdx, oldValue, newValue) => 
+                {
+                    var timer = Timers[hartIdx, timerIdx];
+                    timer.EventEnabled = newValue;
+                }, multiRegValueProviderCallback: (hartIdx, timerIdx) => 
+                {
+                    var timer = Timers[hartIdx, timerIdx];
+                    return timer.EventEnabled;
+                });
+            var intrTestOffset = AddMultiRegisters(registersMap, intrStateOffset, hartId, numberOfTimers,
+                flagPrefix: "IS", regName:"INTR_STATE", multiRegWriteCallback: (hartIdx, timerIdx, oldValue, newValue) => 
+                {
+                    if(newValue) 
+                    {
+                        var gpio = GetIrqForHart(hartIdx, timerIdx);
+                        gpio.Set(false);
+                    }
+                }, multiRegValueProviderCallback: (hartIdx, timerIdx) => 
+                {
+                    var gpio = GetIrqForHart(hartIdx, timerIdx);
+                    return gpio.IsSet;
+                });
+            AddMultiRegisters(registersMap, intrTestOffset, hartId, numberOfTimers,
+                flagPrefix: "T", regName:"INTR_TEST", multiRegWriteCallback: (hartIdx, timerIdx, oldValue, newValue) => 
+                {
+                    if(newValue) 
+                    {
+                        var gpio = GetIrqForHart(hartIdx, timerIdx);
+                        gpio.Set(true);
+                    }
+                }, multiRegValueProviderCallback: (hartIdx, timerIdx) => 
+                {
+                    return false;
+                });
+        }
+
+        private long AddTimerConfigAndValueRegisters(Dictionary<long, DoubleWordRegister> registersMap, long address, int hartId)
+        {
+            long offset = 0;
+            var configRegister = new DoubleWordRegister(this, 0x10000);
+            var i = hartId;
+            configRegister = configRegister.WithValueField(0, 12, out prescalers[hartId], name: $"PRESCALE_{0}", writeCallback: (_, val) => 
```

Comment:
no need to assign back to `configRegister`; in fact you can use a fluent notation here:
```
configRegister
    .WithValueField(...)
    .WithValueField(...)
    .WithValueField(...);
```

Reply:
Shouldn't it be `PRESCALE_{hartId}`?